frontend/src/views/CollabPageView.tsx

```diff
@@ -117,11 +119,53 @@ const CollabPageView: React.FC = () => {
         newSocket.emit("sessionJoined", sessionIdObj, uid);
       });
 
-      newSocket.on("sessionData", ({ sessionIdObj, uid, questionData }) => {
+      newSocket.on("sessionData", async ({ sessionIdObj, uid, questionData }) => {
         sessionIdObj = sessionIdObj;
         // Set state with the received data
         setUserId(uid);
         setQuestionData(questionData);
+
+        const attemptDate = moment().tz("Asia/Singapore").format();
+        setDateAttempted(attemptDate);
+        console.log("Attempt date:", attemptDate);
+
+        const createQuestionAttempted =
+        import.meta.env.VITE_HISTORY_SERVICE_CREATE_QUESTION_BACKEND_URL ||
+        "http://localhost:5005/create-question-attempted";
+
+        console.log("Request URL:", createQuestionAttempted);
+
+        const requestBody = {
+          userUid: uid, 
+          questionUid: questionData.id, 
+          dateAttempted: attemptDate,
+        };
+
+        console.log("Request Body:", requestBody);
+
+        try{
+          const response = await fetch(createQuestionAttempted, {
+            method: "POST",
+            headers: {
+                'Authorization': `Bearer ${token}`, // If using JWT or similar
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(requestBody),
+          });
+
+          console.log("Response Status:", response.status);
+
+          if (response.ok) {
+            const result = await response.json();
+            console.log(result.message);
+          } else {
+              const error = await response.json();
+              console.error('Error creating question:', error.message);
+          }
+        } catch (error) {
+          console.error('Network error:', error);
+        }
+
       });
```

Comment:
Consider using `callFunction` for this

Reply:
oh tq have edited to use callfunction :-)